### PR TITLE
Improve search response classes

### PR DESCRIPTION
**Pull request by dvrylc**
This PR introduces new classes for the response objects of the Students and Instructors search endpoints.

### CODE REVIEW

File: src/main/java/teammates/ui/webapi/output/SearchStudentsStudentData.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    @Nullable
+    private final String comments;
+
+    public SearchStudentsStudentData(StudentAttributes studentAttributes) {
+        super(studentAttributes.getName(), studentAttributes.getEmail(), studentAttributes.getCourse(),
+                studentAttributes.getGoogleId());
+        this.comments = studentAttributes.getComments();
+        this.team = studentAttributes.getTeam();
+        this.section = studentAttributes.getSection();
+        this.joinState = studentAttributes.isRegistered() ? JoinState.JOINED : JoinState.NOT_JOINED;
```
Comment (hcwong) [inline src/main/java/teammates/ui/webapi/output/SearchStudentsStudentData.java]: Think JoinState can be moved to the common class even though it is not used now, reason being just to standardize attributes that can be found in both `StudentData` and `InstructorData` now.

**Review by hcwong — APPROVED** [review]
Just 1 comment, otherwise looks good 

imin

File: src/main/java/teammates/ui/webapi/output/SearchStudentsStudentData.java (hunk 1 above)
Comment (hcwong) [reply src/main/java/teammates/ui/webapi/output/SearchStudentsStudentData.java]: Maybe can also rename the SearchStudentsStudentData to SearchStudentsResult and this file to SearchStudentData to make the file name easier to read. but its not a big deal

Comment (dvrylc) [reply src/main/java/teammates/ui/webapi/output/SearchStudentsStudentData.java]: @hcwong you mean rename `SearchStudentsStudentData` -> `SearchStudentData` and `SearchStudentsData` -> `SearchStudentsResult` right?

Comment (hcwong) [reply src/main/java/teammates/ui/webapi/output/SearchStudentsStudentData.java]: yea soz what was i thinking

**Pull request merged**